### ISSUE TITLE
Add support for multiple clusters for integration tests (CCR)

### ISSUE THREAD

**Issue by dblock** [opensearch-project/opensearch-build#2066]
### Is your feature request related to a problem? Please describe

Coming from https://github.com/opensearch-project/cross-cluster-replication/issues/383#issuecomment-1114747305, CCR needs a multi-node cluster for replication tests. Because of the different scenarios under test with security, CCR ITs create local clusters with specific configurations for each test case. An analysis put that number of different configurations at 7 clusters. This approach was discussed on [this PR](https://github.com/opensearch-project/OpenSearch/pull/1930) where I updated the language around recommendations/practices for testing in OpenSearch.

### Describe the solution you'd like

Need to find a path forward to test at least basic CCR replication functionality for the complete distribution. This could be a "good enough" multi-node cluster available to all plugins to run integration tests.

### Additional context

https://github.com/opensearch-project/opensearch-build/issues/1499
https://github.com/opensearch-project/opensearch-build/issues/1428

**Comment by prudhvigodithi** [comment]
[Triage] Adding @ankitkala, any reason why we cant re-use the performance testing setup that is out there for integration testing?
@peterzhuamazon @gaiksaya

**Comment by ankitkala** [comment]
I think both `LocalTestCluster` and `PerfTestCluster` can be used.  Primary issue is that we only create one test cluster for running the integration tests during the distro build. 
Cross cluster replication requires at least 2 clusters to run the performance tests.

**Comment by bbarani** [comment]
@peterzhuamazon @gaiksaya What would it take to support multiple clusters for CCR integration test? I think we can specifically spin up multiple clusters just for CCR tests as we are running parallely now.

**Comment by peterzhuamazon** [comment]
There are several big changes needs to be done in test framework.
As of now it is assuming only one cluster so discovery is not needed to be configured.
This is not a small task and would require some code diving before we can tell the efforts. Thanks.

**Comment by dblock** [comment]
Practically speaking I would introduce test configuration settings that define the cluster configuration, e.g.:

```yaml
  - name: cross-cluster-replication
    integ-test:
      topology: multi-cluster
      test-configs:
        - with-security
        - without-security
```

or something more elaborate

```yaml
  - name: cross-cluster-replication
    integ-test:
      topology: 
        - name: cluster1
           nodes: 3
        - name: cluster2
           ...
      test-configs:
        - with-security
        - without-security
        - 
```

This is a good opportunity to revisit whether `with-security` and `without-security` was the right approach, and whether we can reuse blocks to avoid the boilerplate, e.g.

```yaml
  - multi-cluster: &multi-cluster
     nodes: 3
  - name: cross-cluster-replication
    integ-test:
      - name: with-security
         security: true
         topology: &multi-cluster
```

**Comment by bbarani** [comment]
@ankitkala Would you be able to create a PR with the above changes? We currently do not have bandwidth to support customized testing process and would really appreciate if you can help implement this functionality.

**Comment by ankitkala** [comment]
I'll be OOO till 17/6. Let me see if I can pick this up after I'm back.

**Comment by ankitkala** [comment]
Needs prioritisation.

**Comment by bbarani** [comment]
@manishdev-amzn can you please prioritize this issue to automat the integration test for CCR?

**Comment by ankitkala** [comment]
@monu-aws  would be working on this.

**Comment by monu-aws** [comment]
Hi @dblock @ankitkala 
Currently LocalTestCluster does not support creating multi-node clusters, also the tests to verify CCR does not depend on the clusters having multiple nodes. I am thinking of keeping the structure of the yml file to be like below
```
- name: cross-cluster-replication
  integ-test:
  topology: multi-cluster
    test-configs:
      - with-security
      - without-security

OR

- name: cross-cluster-replication
  integ-test:
  topology: single-cluster
    test-configs:
      - with-security
      - without-security
```
This can be implemented by creating the two clusters by using LocalTestCluster to create two clusters and pass the endpoints to the integtest.sh script to run the tests.
Is there any other better way to structure the manifest file ?

**Comment by bbarani** [comment]
@ankitkala Whats the current stats of this issue?

**Comment by monu-aws** [comment]
@bbarani, changes are already merged in both CCR and Build repo.
We are tracking this change for 2.6, and will enable the tests for 2.6 immediately after the 2.5 release.

**Comment by gaiksaya** [comment]
Closing as duplicate of https://github.com/opensearch-project/opensearch-build/issues/3448
